documentation/modules/exploit/linux/http/eyesofnetwork_autodiscovery_rce.md

```diff
@@ -0,0 +1,84 @@
+## Introduction
+This module exploits multiple vulnerabilities in EyesOfNetwork version 5.3 and prior in order to execute arbitrary commands as root.
+
+The module first exploits a hardcoded admin API key in EyesOfNetwork API version 2.4.2 (CVE-2020-8657) in order to generate a valid access token and use it to create a new user with admin privileges. If the generated key is not valid, the admin API key is obtained via an SQL injection vulnerability affecting the same API version (CVE-2020-8656).
+
+Next, the module authenticates as the newly created user in order to abuse a command injection vulnerability in the `target` parameter of the AutoDiscovery functionality within the EON web interface (CVE-2020-8654). Specifically, it writes an Nmap NSE script containing the payload to disk, and then activates this script by launching an Nmap host discovery scan against the target. This approach achieves privilege escalation because the default sudo configuration permits the 'apache' user to execute Nmap as root (CVE-2020-8655).
+
+The module only works with HTTPS, so SSL is enabled by default. Valid credentials for a user with administrative privileges are required. However, this module can bypass authentication via two methods, i.e. by generating an API access token based on a hardcoded key, and via SQLI. This module has been successfully tested on EyesOfNetwork 5.3 with API version 2.4.2.
+
+## Verification Steps
+
+1. Install the module as usual
+2. Start msfconsole
+3. Do: `use exploit/linux/http/eyesofnetwork_autodiscovery_rce`
+4. Do: `set RHOSTS [IP]`
+5. Do: `set payload [payload]`
+6. Do: `set LHOST [IP]`
+7. Do: `exploit`
+
+## Options
+
+1.  `Proxies`. This option is not set by default.
+2.  `RHOSTS`. To use: `set RHOSTS [IP]`
+3.  `RPORT`. The default setting is `443`. To use: `set RPORT [PORT]`
+4.  `SERVER_ADDR`. This option is not set by default.
+5.  `SSL`. The default setting is `true`, because HTTPS is required for the module to work.
+6.  `TARGETURI`. This option is the base path. `/` by default.
+7.  `VHOST`. This option is not set by default.
+
+## Payload Options
+1. `LHOST`. To use: `set LHOST [IP]`
+2. `LPORT`. The default setting is `4444`. To use: `set LPORT [PORT]`
+
```

Comment:
We typically don't need information about the more common options like RHOSTS, RPORT, SSL, TARGETURI, etc. Most of these options can be removed unless they're unique to this specific module.